### DISCUSSION TITLE
Override dependency on init

### DISCUSSION THREAD

**Discussion by PierreCapo** [pointfreeco/swift-composable-architecture#1744] [Q&A] [answered]
Context is the same as https://github.com/pointfreeco/swift-composable-architecture/discussions/1613 where some state needs to be instantiated by a dependency.

The problem mentioned in the discussion talks about the ability to override a dependency in `TestStore`. However, in some tests we also want to test against a `Store` implementation, for instance when using the https://github.com/pointfreeco/swift-snapshot-testing library.

To sum up, given a reducer that needs a dependency on init :

```Swift
// SettingsCore.swift
struct SettingsCore {
  struct State {
    var localAuthType: LABiometryType
    
    init() {
      @Dependency(\.localAuthClient) var localAuthClient
        
       self.localAuthType = localAuthClient.getType()
    }
  }
}
```

```Swift
// SettingsTests.swift
func testSettingsScreen() {
    let screen = SettingsView(
            store: .init(
                initialState: .init(),
                reducer: EmptyReducer()
            )
        )

    assert(snapshot: screen)
}
```

What is the way to override this dependency also on init in the test suite?

**Comment by mbrandonw** [comment, marked as answer]
Hi @PierreCapo, interesting use case. It does seem that if we support overriding dependencies upon initialization in `TestStore`, then `Store` should probably also support it.

We will have to think about it a bit more, but in the meantime you can do this:

```swift
func testSettingsScreen() {
  DependencyValues.withValues {
    // override
  } operation: {
    let screen = SettingsView(
      store: .init(
        initialState: .init(),
        reducer: EmptyReducer()
      )
    )

    assert(snapshot: screen)
  }
}
```

**Reply by PierreCapo** [reply]
Many thanks @mbrandonw, this works great as a workaround 🎉

Side note, I don't find this API in the documentation, do you want that I open a PR to add an extra paragraph about this pattern in [the dependency management documentation](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/dependencymanagement) ?

**Reply by mbrandonw** [reply]
Hi @PierreCapo, that API doesn't show in the docs because docc doesn't support hosting docs for multiple libraries in the same package, and technically all the dependency stuff is in the `Dependencies` module. We have plans to release `Dependencies` as its own library soon and so that will help the doc situation.

But also I think you suggestion of describing the pattern in the TCA docs is good. We will try to do that soon. Thanks!
